dnf-behave-tests/features-createrepo_c/createrepo_c_empty_repository.feature

```diff
@@ -0,0 +1,396 @@
+  And repodata in "/temp-repos/temp-repo/repodata/" is
+     | Type         | File                                                                                  | Checksum Type | Compression Type |
+     | primary      | ${checksum}-primary.xml.gz                                                            | sha256        | gz               |
+     | filelists    | ${checksum}-filelists.xml.gz                                                          | sha256        | gz               |
+     | other        | ${checksum}-other.xml.gz                                                              | sha256        | gz               |
+     | primary_db   | ${checksum}-primary.sqlite.bz2                                                        | sha256        | bz2              |
+     | filelists_db | ${checksum}-filelists.sqlite.bz2                                                      | sha256        | bz2              |
+     | other_db     | ${checksum}-other.sqlite.bz2                                                          | sha256        | bz2              |
+  And repodata "/temp-repos/temp-repo/repodata/" are consistent
+  And file "/temp-repos/temp-repo/repodata/repomd.xml" contains lines
+  """
+  <tags>
+  \s+<repo>repotag_a</repo>
+  \s+<repo>repotag_b</repo>
+  </tags>
+  """
+
+
+Scenario: Repo from empty directory with --revision XYZ
+ When I execute createrepo_c with args "--revision XYZ ./temp-repo" in directory "/temp-repos"
+ Then the exit code is 0
+  And repodata in "/temp-repos/temp-repo/repodata/" is
+     | Type         | File                                                                                  | Checksum Type | Compression Type |
+     | primary      | ${checksum}-primary.xml.gz                                                            | sha256        | gz               |
+     | filelists    | ${checksum}-filelists.xml.gz                                                          | sha256        | gz               |
+     | other        | ${checksum}-other.xml.gz                                                              | sha256        | gz               |
+     | primary_db   | ${checksum}-primary.sqlite.bz2                                                        | sha256        | bz2              |
+     | filelists_db | ${checksum}-filelists.sqlite.bz2                                                      | sha256        | bz2              |
+     | other_db     | ${checksum}-other.sqlite.bz2                                                          | sha256        | bz2              |
+  And repodata "/temp-repos/temp-repo/repodata/" are consistent
+  And file "/temp-repos/temp-repo/repodata/repomd.xml" contains lines
+  """
+  <revision>XYZ</revision>
+  """
+
+
+Scenario: Repo from empty directory with --skip-symlinks and symlinked package
+ Then I create symlink from "{context.dnf.repos_location}/dnf-ci-fedora/noarch/setup-2.12.1-1.fc29.noarch.rpm" to "/temp-repos/temp-repo/setup-2.12.1-1.fc29.noarch.rpm"
```

Comment:
Applies to all the feature files, I've just picked this one instance.

As I've said before it's kind of a misfeature of behave to allow basically any combination of `{Given,When,Then} something` (although we could alleviate that by defining the steps better perhaps), while at the same time the semantics often have to depend on the author, for:

`Given` - test setup, you shouldn't have to test it was successful
`When` - run the command that is being tested
`Then` - validate the result of the test

`And` - supposedly used as a continuation of one of the three above steps, doesn't really make things easier especially when you move lines around :slightly_frowning_face: 

At least that's how I understand it... Anyway, if you wouldn't mind fixing it throughout :slightly_smiling_face: